src/main/scala/firrtl/stage/FirrtlStage.scala

```diff
@@ -0,0 +1,38 @@
+// See LICENSE for license details.
+
+package firrtl.stage
+
+import firrtl.{AnnotationSeq, CustomTransformException, FIRRTLException, Utils}
+import firrtl.options.{Stage, Phase, PhaseException, Shell, OptionsException}
+import firrtl.options.phases.DeletedWrapper
+import firrtl.passes.{PassException, PassExceptions}
+
+import scala.util.control.ControlThrowable
+
+import java.io.PrintWriter
+
+class FirrtlStage extends Stage {
```

Comment:
Should this be an object?
```
$ java -cp utils/bin/firrtl.jar firrtl.stage.FirrtlStage -i regress/ICache.fir -o out -X verilog
Error: Main method is not static in class firrtl.options.Stage, please define the main method as:
   public static void main(String[] args)
```

Interestly, and I don't pretend to know why this worked, but:
```
$ java -jar utils/bin/firrtl.jar firrtl.stage.FirrtlStage -i regress/ICache.fir -o out -X verilog                        
------------------------------------------------------------------------------
Warning: firrtl.Driver is deprecated since 1.2!
Please switch to firrtl.stage.FirrtlStage
------------------------------------------------------------------------------
Total FIRRTL Compile Time: 1081.4 ms
]
```

Reply:
I changed everything to classes so that the dependencies could be defined as `Class[A <: DependencyAPI[A]]` (see: https://github.com/freechipsproject/firrtl/pull/1015/files#diff-8cd99845a8f2c996dfdce4295c252b3cR26).

I think the clean approach is to add an object that constructs a `FirrtlStage`.

Reply:
Current solution: 
  - Move the `main` method inside `Stage` into a new class `StageMain`
  - Add an `object FirrtlMain extends StageMain(new FirrtlStage)`
  - Change name of `FirrtlStageSpec` to `FirrtlMainSpec` and update this to use `FirrtlMain`